From 5ed90106f49855b29da6c7cc07c8e799654d85c4 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Fri, 7 Apr 2017 01:04:48 +0200
Subject: [PATCH] Input: edt-ft5x06: Add fake pressure reporting

Report binary pressure to satisfy requirements of broken userspace.
For example tslib test tools need this pressure reporting.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 drivers/input/touchscreen/edt-ft5x06.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/input/touchscreen/edt-ft5x06.c b/drivers/input/touchscreen/edt-ft5x06.c
index 240e8de24cd2..cd6063b6845d 100644
--- a/drivers/input/touchscreen/edt-ft5x06.c
+++ b/drivers/input/touchscreen/edt-ft5x06.c
@@ -252,6 +252,8 @@ static irqreturn_t edt_ft5x06_ts_isr(int irq, void *dev_id)
 					       type != TOUCH_EVENT_UP))
 			touchscreen_report_pos(tsdata->input, &tsdata->prop,
 					       x, y, true);
+		input_report_abs(tsdata->input, ABS_MT_PRESSURE,
+				 type != TOUCH_EVENT_UP);
 	}
 
 	input_mt_report_pointer_emulation(tsdata->input, true);
@@ -1143,6 +1145,7 @@ static int edt_ft5x06_ts_probe(struct i2c_client *client,
 		input_set_abs_params(input, ABS_MT_POSITION_Y,
 				     0, 65535, 0, 0);
 	}
+	input_set_abs_params(input, ABS_MT_PRESSURE, 0, 1, 0, 0);
 
 	touchscreen_parse_properties(input, true, &tsdata->prop);
 
-- 
2.25.1

